### ISSUE TITLE
Sway should throw a better defined error when there is an invalid `$ref`

### ISSUE THREAD

**Issue by saharj** [apigee-127/sway#62]
Code Sample:

```
swagger: '2.0'
info:
  title: Example
  description: Example
  version: 1.0.0
host: 'example.com'
basePath: /
schemes:
  - http
produces:
  - application/json
paths:
  /rest_auth:
    $ref: '#/x-paths-definitions/auth_index'
  /rest_auth/login:
    post:
      $ref: '#/x-paths-definitions/auth_index/post'
x-paths-definitions:
  auth_index:
    post:
      tags:
        - Auth
      summary: Logs the user in
      description: |
        Tests a pair of user_id and password to log the user in.
        <br /><br />
        Aliases: /rest_auth/login, /rest_auth
      parameters:
        - name: body
          in: body
          description: The user_id and password of the user trying to login.
          required: true
          schema:
            required:
              - user_id
              - password
            properties:
              user_id:
                type: string
              password:
                type: string
            type: object
      responses:
        '200':
          description: User object of the newly logged-in user + user_cd + blog_cd
          schema:
            properties:
              user:
                type: object
              user_cd:
                type: integer
              blog_cd:
                type: integer
            type: object
```

Please see https://github.com/swagger-api/swagger-editor/issues/756

**Comment by whitlockjc** [comment]
The problem right now is that swagger-tools and sway allow you to have `$ref` anywhere because we dereference references prior to doing validation.  One the one hand, this is very nice.  On the other, it does allow you to have non-valid Swagger documents prior to dereferencing regardless of whether they are valid after dereferencing.  I will update the validation of references to make sure they are only in the allowed locations but I am also hoping that future versions of Swagger do not impose this limitation as I think it's an oversight.

**Comment by whitlockjc** [comment]
This was fixed in `json-refs@v3.0.4` and commited in https://github.com/apigee-127/sway/commit/cdd8030a7dac5f86de5fb0bdf719b7ed92ada1f0
